### ISSUE TITLE
an empty "data_table" file is required for standalone runs 

### ISSUE THREAD

**Issue by DeniseWorthen** [ufs-community/ufs-weather-model#652]
## Description

For the coupled or datm applications,  the ``data_table`` file provides the name and use information for the river runoff data file. It is applied in MOM6 using the data_override function. For standalone ATM tests, an empty data_file is required to be present in the run directory or else the run fails. 

### To Reproduce:

Remove the empty data_table file from a standalone FV3 run directory, the run will fail.

## Additional context

The cause is the call to `` data_override_init`` in ``module_fcst_grid_comp.F90``.  Testing with hera intel/gnu shows that removing this call (and the associated ``use data_override_mod,  only: data_override_init`` ) does not change baselines. The  empty ``data_table`` file is not required at run time.

**Comment by climbfuji** [comment]
It would be nice to get rid of this empty file, thanks for digging through the code.

**Comment by junwang-noaa** [comment]
Currently data_table in fv3atm is empty, not sure if the non-empty data_table will be used in certain cases. Let's check with Rusty to confirm there is no impact to remove the data_table in fv3atm.

**Comment by DeniseWorthen** [comment]
We could also implement an inquire call for the data_table and try to initialize data_override only if data_table exists.

**Comment by junwang-noaa** [comment]
You mean the data_table is not a empty file? I think this is better as it allows developers to run the model with non-empty data_table.

**Comment by DeniseWorthen** [comment]
When we use MOM6, we have a data_table present in the run directory to specify runoff. For the uncoupled case, we could implement the data_override_init in FV3 like:

```
   inquire(FILE=data_table, EXIST=fexist)
   if (fexist ) then
     call data_override_init ( ) ! Atm_domain_in  = Atm%domain, &
                                                ! Ice_domain_in  = Ice%domain, &
                                                ! Land_domain_in = Land%domain )
   endif
```
Then at least an empty data_table would not be required in the run directory unless some standalone test actually needed it.

When it is coupled, the data_table for ocean would I guess trigger this initialization, but since the line starts with "OCN", FMS is smart enough to figure out that it is only implementing the override for the ocean:

```
"OCN", "runoff", "runoff", "./INPUT/FRUNOFF", "none" ,  1.0
```

**Comment by climbfuji** [comment]
Would it make sense and is it possible to allow this bypass only when `data_table` is not required, i.e. when currently an empty file is supposed to be there?

**Comment by DeniseWorthen** [comment]
I don't think there is any reason that an empty data_table is supposed to be there, other than we're requiring it to be there by asking FMS to initialize the data_override function. 

!! Given a gridname, fieldname and model time this routine will get data in a file whose
!! path is described in a user-provided data_table, do spatial and temporal interpolation if
!! necessary to convert data to model's grid and time.

**Comment by climbfuji** [comment]
> I don't think there is any reason that an empty data_table is supposed to be there, other than we're requiring it to be there by asking FMS to initialize the data_override function.
> 
> !! Given a gridname, fieldname and model time this routine will get data in a file whose
> !! path is described in a user-provided data_table, do spatial and temporal interpolation if
> !! necessary to convert data to model's grid and time.

Ok, I don't understand in full how this initialization in MOM6 will work, but as long as the model aborts in case there should be a non-empty `data_table` file, then all good.

**Comment by DeniseWorthen** [comment]
MOM6 itself calls data_override_init (and data_override) if the parameter settings indicate it wants to use the function to provide some form of "data"---runoff or fluxes.

**Comment by junwang-noaa** [comment]
I think it is good to add the inquire if statement around the call of data_overide_init.

**Comment by DeniseWorthen** [comment]
Will be fixed in PR #654
